packages/language-server/src/core/documents/AstroDocument.ts

```diff
@@ -29,7 +29,12 @@ export class AstroDocument extends WritableDocument {
 		this.updateDocInfo();
 	}
 
-	getText(): string {
+	getText(range?: Range | undefined): string {
+		if (range) {
+			const start = this.offsetAt(range.start);
+			const end = this.offsetAt(range.end);
+			return this.content.substring(start, end);
+		}
```

Comment:
Despite us not using this, it's necessary that we support it because `vscode-html-languageservice` uses that feature to get the name of the hovered HTML attribute (despite having access to it in much simpler ways in my opinion but oh well) for hover info